tracing/log.go

```diff
@@ -0,0 +1,68 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package tracing
+
+import (
+	"github.com/sirupsen/logrus"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/trace"
+)
+
+// NewLogrusHook creates a new logrus hook
+func NewLogrusHook() *LogrusHook {
+	return &LogrusHook{}
+}
+
+// LogrusHook is a logrus hook which adds logrus events to active spans.
+// If the span is not recording or the span context is invalid, the hook is a no-op.
+type LogrusHook struct{}
+
+// Levels returns the logrus levels that this hook is interested in.
+func (h *LogrusHook) Levels() []logrus.Level {
+	return logrus.AllLevels
```

Comment:
How about making the level customizable? Most containerd users may not need `debug` level information.

Reply:
Sorry. I'm unsure this is the place to make the level customizable. If containerd is configured to log events above the info level, would Open Telemetry follow that as well?

Reply:
Level is configured by the Logrus logger itself (e.g. logrus.SetLevel)
Hooks only fired for levels that logrus is actually configured to emit.

Reply:
Example test:

```go
package main

import (
	"fmt"

	"github.com/sirupsen/logrus"
)

func main() {
	var h hook
	logrus.AddHook(h)

	logrus.Info("TestInfo")
	logrus.Debug("TestDebug")

}

type hook struct {
}

func (hook) Fire(e *logrus.Entry) error {
	fmt.Println("Fired hook for", e.Level)
	return nil
}

func (hook) Levels() []logrus.Level {
	return logrus.AllLevels
}
```

Outputs:

```
Fired hook for info
INFO[0000] TestInfo
```

Reply:
Thanks! That's the only question I have.